wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -78,18 +83,25 @@ sealed abstract class Wallet extends UnlockedWalletApi with BitcoinSLogger {
       outPoint: TransactionOutPoint,
       addressDb: AddressDb): Future[UTXOSpendingInfoDb] = {
 
-    val utxo = UTXOSpendingInfoDb(
-      id = None,
-      outPoint = outPoint,
-      output = output,
-      privKeyPath = addressDb.path,
-      redeemScriptOpt = None, // todo fix me when implementing P2SH addresses
-      scriptWitnessOpt = addressDb.witnessScriptOpt
-    )
-
-    utxoDAO.create(utxo).map { _ =>
-      logger.trace(
-        s"Successfully inserted UTXO ${outPoint.txId.hex}:${outPoint.vout.toInt} into DB")
+    val utxo: UTXOSpendingInfoDb = addressDb match {
+      case segwitAddr: SegWitAddressDb =>
+        SegWitUTOXSpendingInfodb(
+          id = None,
+          outPoint = outPoint,
+          output = output,
+          privKeyPath = segwitAddr.path,
+          scriptWitness = segwitAddr.witnessScript
+        )
+      case badCombination =>
```

Comment:
Is this just unimplemented for now? If so, i'd rather just explicitly match and use `???`

Reply:
explicitly matched a couple of places